### PR TITLE
feat: add support for consensys zkevm network (Linea)

### PR DESCRIPTION
**Pull request by VGau**
**PR Title**
Added support for ConsenSys zkEVM test network (Linea).

**Description**
It was added ConsenSys zkEVM test network (Linea) support to our Network Controller.

- CHANGED:
  - The network controller has been updated to support Consensys zkEVM (Linea)
  - New test cases have been added to use ConsenSys zkEVM network (Linea)

**Checklist**

- [X] Tests are included if applicable
- [X] Any added code is fully documented

### CODE REVIEW

**Review by sethkfman — DISMISSED** [review]
LGTM

File: packages/network-controller/tests/NetworkController.test.ts [hunk 1]
```diff
@@ -643,6 +643,224 @@ describe('NetworkController', () => {
         });
       });
 
+      describe(`when the provider config in state contains a network type of "linea testnet"`, () => {
```
Comment (mcmire) [inline packages/network-controller/tests/NetworkController.test.ts:646]: Typo: `"linea testnet"` -> `"lineatestnet"`

**Review by mcmire — COMMENTED** [review]
This makes sense. Looks like there's one block of tests missing though. I think you'll want to copy the `describe` starting at line 2222 to a new one and change instances of `"localhost"` to `"lineatestnet"`.

**Review by sethkfman — APPROVED** [review]
Reviewed typo. LGTM

**Comment by sethkfman** [comment]
@mcmire Can you confirm?

**Review by mcmire — DISMISSED** [review]
Yes, LGTM!

**Review by mcmire — APPROVED** [review]

**Comment by mcmire** [comment]
@Gudahtt I notice you left a comment in Slack about this PR, are you good with this?

**Comment by VGau** [comment]
> @Gudahtt I notice you left a comment in Slack about this PR, are you good with this?

Yes I am good with this PR thanks 😄

**Comment by Gudahtt** [comment]
Nope, no objections. But I will add a `DO-NOT-MERGE` label for now because since this was approved, the strategy used to support this network on mobile has changed. We need to revisit whether we need a change to the network controller at all, or whether this can be handled solely in mobile instead.

**Comment by mcmire** [comment]
Now that https://github.com/MetaMask/core/pull/1423 has been merged, can we close this?

**Comment by Gudahtt** [comment]
I'm assuming we don't need this anymore; if it turns out we do, we can open a new PR.

**Pull request closed without merging**